### DISCUSSION TITLE
Integrated device parameter check

### DISCUSSION THREAD

**Discussion by KrzysztofHerman** [ra3xdh/qucs_s#685] [Ideas]
Would it be possible to implement some integrated device parameter check to validate if the introduced value is within a certain scope. An example could be MOS transistor sizing, where W and L are the design variables. The checker should inform the designer if W or L are  below or either above a certain limit (Wmin, Wmax, Lmin, Lmax). The max/min parameters could be the part of the XML-like file which defines the device symbol.  
For example in the IHP-Open-PDK the mosfet device can be instantiated using this spice code 
` X4 _net1  _net0  _net2  _net3 sg13_lv_nmos w=400u l=0.5u`
The netlist simulates however provide wrong values due to the fact that maximum width is equal to 10u. In other words the model is valid up to 10u. The implementation of the checks would prevent the designer to get into such kind of situations and force him to properly declare the device: 
` X4 _net1  _net0  _net2  _net3 sg13_lv_nmos w=400u l=0.5u ng=40`

**Comment by ra3xdh** [comment]
Adding parameter limits check for system or library devices is technically possible. It's need to implement some methods/fields in `Property` and `Component` classes and take care of backward compatibility. But I cannot say when this feature could be added in Qucs-S.
